catalyst/core/callbacks/pruning.py

```diff
@@ -0,0 +1,186 @@
+                    "global pruning is not currently support."
+                )
+            if "unstructured" not in pruning_fn:
+                if dim is None:
+                    raise Exception(
+                        "If you are using structured pruning you"
+                        "need to specify dim in callback args"
+                    )
+                if pruning_fn == "ln_structured":
+                    if n is None:
+                        raise Exception(
+                            "If you are using ln_unstructured you"
+                            "need to specify n in callback args"
+                        )
+                    self.pruning_fn = self._wrap_pruning_fn(
+                        prune.ln_structured, dim=dim, n=n
+                    )
+                else:
+                    self.pruning_fn = self._wrap_pruning_fn(
+                        PRUNING_FN[pruning_fn], dim=dim
+                    )
+            else:  # unstructured
+                self.pruning_fn = PRUNING_FN[pruning_fn]
+        else:
+            self.pruning_fn = pruning_fn
+        if keys_to_prune is None:
+            keys_to_prune = ["weight"]
+        self.prune_on_epoch_end = prune_on_epoch_end
+        self.prune_on_stage_end = prune_on_stage_end
+        if not (prune_on_stage_end or prune_on_epoch_end):
+            warnings.warn(
+                "Warning!"
+                "You disabled pruning pruning both on epoch and stage end."
+                "Model won't be pruned by this callback."
+            )
+        self.remove_reparametrization = remove_reparametrization
+        self.keys_to_prune = keys_to_prune
+        self.amount = amount
+        self.reinitialize_after_pruning = reinitialize_after_pruning
+        self.layers_to_prune = layers_to_prune
+
+    @staticmethod
+    def _wrap_pruning_fn(pruning_fn, *args, **kwargs):
+        return lambda module, name, amount: pruning_fn(
+            module, name, amount, *args, **kwargs
+        )
+
+    def _prune_module(self, module):
+        for key in self.keys_to_prune:
+            self.pruning_fn(module, name=key, amount=self.amount)
+
+    def _to_be_pruned(self, layer_name):
+        return (
+            self.layers_to_prune is None or layer_name in self.layers_to_prune
+        )
+
+    def _run_pruning(self, model: torch.nn.Module):
+        pruned_modules = 0
+        for name, module in model.named_modules():
+            try:
+                if self._to_be_pruned(name):
+                    self._prune_module(module)
+                    pruned_modules += 1
+            except AttributeError as e:
+                if self.layers_to_prune is not None:
+                    raise e
+
+        if pruned_modules == 0:
+            raise Exception(
+                f"There is no {self.keys_to_prune} key in your model"
+            )
+        if self.reinitialize_after_pruning:
+            model.apply(weight_reset)
```

Comment:
could we make this function static, or just move it to catalyst.utils?
for example, make it:
```python
def prune_model(model, layers_to_prune: optional[str] = None, reinitialize_after_pruning=False, keys_to_prune, amount): 
   ...
```
?